### PR TITLE
Normalize kernels

### PR DESCRIPTION
**Pull request by stoprightthere**
Normalize kernels and feature maps, with a flag to turn off this behaviour.

### CODE REVIEW

File: tests/spaces/test_graph.py [hunk 1]
```diff
@@ -105,14 +105,14 @@ def run_tests_with_adj(A, L, tol=1e-14, tol_m=1e-7):
...
-    K_cons = MaternKarhunenLoeveKernel(graph, n)
+    K_cons = MaternKarhunenLoeveKernel(graph, n, normalize=False)
     params, state = K_cons.init_params_and_state()
     idx = B.cast(B.dtype(A), np.arange(n)[:, None])
 
     nu = B.cast(B.dtype(A), np.array([1.0]))
     lscale = B.cast(B.dtype(A), np.array([1.0]))
 
-    K_normed_cons = MaternKarhunenLoeveKernel(normed_graph, n)
+    K_normed_cons = MaternKarhunenLoeveKernel(normed_graph, n, normalize=False)
```
Comment (vabor112) [inline tests/spaces/test_graph.py:115]: The "normed" in `K_normed_cons` (that refers to the normalized Laplacian) conflicts somewhat with the `normalize` that refers to the normalization constant. Maybe use something like `K_nlap_cons` instead?

File: geometric_kernels/kernels/feature_maps.py
Comment (vabor112) [inline geometric_kernels/kernels/feature_maps.py:1]: Shouldn't we add normalization to `deterministic_feature_map_compact` and `random_phase_feature_map_compact` as well?

**Review by vabor112 — COMMENTED** [review]
Please take a look at my comments.

Beyond them, how about we add normalization tests: for `DiscreteSpectrumSpace` use the space's uniform sampling method to get a bunch of xs and check that the the average of k(x, x) is close to 1. For noncompact, nonuniformly sample xs and check that k(x,x) = 1 for each.

**Comment by vabor112** [comment]
BTW, you intended to drop Python 3.7 support but here it is still present and still causes failed checks.

File: geometric_kernels/kernels/feature_maps.py
Comment (stoprightthere) [reply geometric_kernels/kernels/feature_maps.py:1]: But it's in the kernel?

**Comment by stoprightthere** [comment]
Not only did I "intend" -- I did drop py3.7, check out the devel branch. Once merged, only py>=3.8 checks will be performed.
Re: tests, valid point.

File: geometric_kernels/kernels/feature_maps.py
Comment (vabor112) [reply geometric_kernels/kernels/feature_maps.py:1]: IMHO this is confusing from the user experience perspective, let's discuss this

**Review by vabor112 — CHANGES_REQUESTED** [review]
Pls implement the discussed changes.

**Pull request merged**